### ISSUE TITLE
Deprecate DryRunVisitor.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6333]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)| @tomprince reported|
|-|-|
|Trac ID|trac#6333|
|Type|enhancement|
|Created|2013-02-26 23:59:38Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-dryrunvisitor-6333-2|

It is no longer used ([#4063](https://github.com/twisted/twisted/issues/4063)) and the support for visitors is being removed ([#3231](https://github.com/twisted/twisted/issues/3231)), so there is no reason to keep this around.

<details><summary>Searchable metadata</summary>

```
trac-id__6333 6333
type__enhancement enhancement
reporter__tom_prince tom.prince
priority__normal normal
milestone__ 
branch__branches_deprecate_dryrunvisitor_6333_2 branches/deprecate-dryrunvisitor-6333-2
branch_author__tomprince tomprince
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__easy easy
time__1361923178000000 1361923178000000
changetime__1363631326000000 1363631326000000
version__None None
owner__tom_prince tom.prince
cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_2"></a>|@tomprince commented|
|-|-|

(In [37274]) Branching to deprecate-dryrunvisitor-6333.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_3"></a>|@tomprince commented|
|-|-|

This should probably land after [#3231](https://github.com/twisted/twisted/issues/3231).

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_4"></a>|@tomprince commented|
|-|-|

[build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/deprecate-dryrunvisitor-6333)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph set owner to @tomprince|
|-|-|

This looks fine to me, such as it is, but, as with many deprecations in Twisted, I feel like the explanation is not really sufficient.  Why are visitors no longer supported? If I had written some code (what code might I have written?) that was using this class, what do I need to do instead of using visitors?  If you can think of more useful words to add to the deprecation message explaining why this was done, please do so.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

> This looks fine to me, such as it is, but, as with many deprecations in Twisted, I feel like the explanation is not really sufficient.

I agree.

> Why are visitors no longer supported? If I had written some code (what code might I have written?) that was using this class, what do I need to do instead of using visitors? If you can think of more useful words to add to the deprecation message explaining why this was done, please do so.

Visitors were deprecated in [#2897](https://github.com/twisted/twisted/issues/2897), to be replaced by test decorators.  I think jml or lifeless should be herded into a closet until they write up some documentation about test decorators.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_7"></a>|@tomprince commented|
|-|-|

(In [37586]) Branching to deprecate-dryrunvisitor-6333-2.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_8"></a>|@tomprince set status to `closed`|
|-|-|

(In [37587]) Merge deprecate-dryrunvisitor-6333-2: Deprecate DryRunVisitor.

Author: tom.prince
Reviewers: glyph
Fixes: [#6333](https://github.com/twisted/twisted/issues/6333)

It is no longer used ([#4063](https://github.com/twisted/twisted/issues/4063)) and the support for visitors is
being removed ([#3231](https://github.com/twisted/twisted/issues/3231)), so there is no reason to keep this
around.
